### ISSUE TITLE
Primary UserID

### ISSUE THREAD

**Issue by tobiasjungnickel** [krzyzanowskim/ObjectivePGP#113]
If one key has multiple UserIDs, is it safe that the last key is always the primary one?

`ObjectivePGP.readKeys(fromPath: path).first!.secretKey!.users.last!`

While, for one single key file, above is true for me using secretKey, it isn't the case for publicKey which always have the same UserID order regardless which one is set as primary.

**Comment by krzyzanowskim** [comment]
no. The method isn't part of the public API, but `-[PGPPartialKey primaryUserAndSelfCertificate:]` has the logic.

Thanks for the feedback though, it's probably a good idea to have this information part of the public API.

**Comment by tobiasjungnickel** [comment]
Thanks for adding it to public API

**Comment by krzyzanowskim** [comment]
Looks good in Playground.

<img width="1100" alt="screen shot 2018-05-03 at 00 32 44" src="https://user-images.githubusercontent.com/758033/39552435-88b97e84-4e69-11e8-91ad-90f56246dbca.png">

If you use binary framework, you may need to build it for yourself from master

**Comment by tobiasjungnickel** [comment]
I'm using CocoaPods to get that commit
`pod 'ObjectivePGP', :git => 'https://github.com/krzyzanowskim/ObjectivePGP.git', :commit => '081d543'`
Still only getting 0.11.0. Does it have to be updated in the podspec?

**Comment by krzyzanowskim** [comment]
yes. I only build binary frameworks for podspec when release new version

**Comment by tobiasjungnickel** [comment]
@krzyzanowskim As i now started to use primaryUser i'm receiving strange results out of it. I attached 4 keys with 2 being the same public/private keypairs and 2 being the same public only key. The difference between both of each same keys is that i switched the primary user.
- ...PublicUserID1...: public key with userID1 being primary
- ...PublicUserID2...: public key with userID2 being primary
- ...SecretUserID1...: public/private key with userID1 being primary
- ...SecretUserID2...: public/private key with userID2 being primary

While primary user seems to be switched for public/private key, public only key return userID2 for both.

**Comment by krzyzanowskim** [comment]
> i switched the primary users.

what do you mean by that. How did you switch the primary user?

> While primary user seems to be switched for public/private key, public only key return userID2 for both.

I'm not sure If I follow, the expected is:
KeyMultipleUserIDsPublicUserID1Primary.asc -> UserID1 <userid1@key.de>
KeyMultipleUserIDsSecretUserID1Primary.asc -> UserID1 <userid1@key.de>
amarit?

**Comment by tobiasjungnickel** [comment]
i switched the primary userID by using GPG Keychain from https://gpgtools.org

your supposed expection is correct. The keys should return the primary user as it is in the name of the file UserID1 <> UserID2

**Comment by tobiasjungnickel** [comment]
did you already had time to check if you encounter the same situation or lot of other ToDos from WWDC?

**Comment by krzyzanowskim** [comment]
I was able to reproduce. I'll find some time to fix it in few days.

**Comment by tobiasjungnickel** [comment]
For me checking primaryUser of secretKey is valid, but it fails for publicKey.

**Comment by krzyzanowskim** [comment]
hm... how exactly you check it. Is it reproducible with the test https://github.com/krzyzanowskim/ObjectivePGP/blob/9c9369896602f34e89402bae863c3ecdd8e1b163/Tests/PGPTests.m#L432 ?
